crates/mod-crypto/src/lib.rs

```diff
@@ -0,0 +1,375 @@
+use anyhow::anyhow;
+use aws_lc_rs::cipher::{
+    AES_128, AES_256, DecryptionContext, PaddedBlockDecryptingKey, PaddedBlockEncryptingKey,
+    UnboundCipherKey,
+};
+
+use data_encoding::{
+    BASE32, BASE32HEX, BASE32HEX_NOPAD, BASE32_NOPAD, BASE64, BASE64URL, BASE64URL_NOPAD,
+    BASE64_NOPAD, HEXLOWER,
+};
+use config::{any_err, from_lua_value, get_or_create_sub_module};
+use mlua::prelude::LuaUserData;
+use mlua::{Lua, MetaMethod, UserDataFields, UserDataMethods, Value, Error as LuaError};
+use data_loader::KeySource;
+use hex::decode;
+use std::str;
+
+use serde::Deserialize;
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct AesParams {
+    pub algorithm: AesAlgo,
+    pub key: AesKey,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub enum AesAlgo {
+    Ecb(),
+    Cbc(),
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct LuaCfg {
+    pub key: KeySource,
+    pub value: String,
+    pub algorithm: String,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct LuaCfgDecrypt {
+    pub key: KeySource,
+    pub value: Vec<u8>,
+    pub algorithm: String,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub enum AesKey {
+    Aes128([u8; 16]),
+    Aes256([u8; 32]),
+}
+
+fn is_hex(vec: &[u8]) -> bool {
+    let hex_str = String::from_utf8_lossy(vec);
+    hex_str.chars().all(|c| c.is_digit(16)) && hex_str.len() % 2 == 0
+}
+
+impl AesKey {
+    pub fn from_bytes(bytes: &[u8]) -> Result<Self, String> {
+        let decoded_bytes = if is_hex(bytes) {
+            decode(bytes).map_err(|e| format!("Failed to decode hex: {}", e))?
+        } else {
+            bytes.to_vec()
+        };
+        match decoded_bytes.len() {
+            16 => Ok(AesKey::Aes128(decoded_bytes.try_into().unwrap())),
+            32 => Ok(AesKey::Aes256(decoded_bytes.try_into().unwrap())),
```

Comment:
the `unwrap` calls here will cause a panic, which is undesirable when we can raise an error instead.  Can we simply use the `?` operator here to propagate the error?